Source/GameView/Repositories/GameDetailsRepository/GameDetailsAPIRepo.swift

```diff
@@ -0,0 +1,175 @@
+//  Created by Tiewhan Smith on 2020/04/22.
+//
+
+import Foundation
+
+class GameDetailsAPIRepo {
+  
+  struct gameDataBundleKeys {
+    static let game: String = "game"
+  }
+
+  struct gameDetailsJSONResponseKeys {
+    
+    static let data: String = "data"
+    static let priceOverview: String = "price_overview"
+    static let finalPriceOverview: String = "final"
+    static let shortDescription: String = "short_description"
+    static let developers: String = "developers"
+    static let publishers: String = "publishers"
+    static let headerImage: String = "header_image"
+    
+  }
+
+  public weak var observer: GameDetailsRepoObserver?
+  
+  public init() { }
+  
+}
+
+extension GameDetailsAPIRepo: GameDetailsRepositoryType {
+  
+  public func getGameDetails(of gameList: [Game], with serviceCaller: ServiceCaller) {
+    
+    let operationQueue = OperationQueue()
+    
+    let serviceCallOperation = createServiceCallBlockOperation(using: serviceCaller,
+                                                               forGamesList: gameList)
+    
+    let notifyOperation = BlockOperation {
+      self.notifyGameDetailsLoaded(withData: gameList)
+    }
+    
+    notifyOperation.addDependency(serviceCallOperation)
+    
+    operationQueue.addOperation(serviceCallOperation)
+    operationQueue.addOperation(notifyOperation)
+    
+  }
+  
+  public func subscribeToRepository(subscriber observer: GameDetailsRepoObserver) {
+    self.observer = observer
+  }
+  
+  public func unsubscribeFromRepository() {
+    observer = nil
+  }
+  
+  private func createServiceCallBlockOperation(using serviceCaller: ServiceCaller,
+                                               forGamesList gameList: [Game]) -> BlockOperation {
+    
+    let serviceCallOperation = BlockOperation {
+      
+      let dispatchGroup = DispatchGroup()
+      
+      gameList.forEach({ (game) in
+        
+        dispatchGroup.enter()
+        
+        let dataBundle = DataBundle()
+        dataBundle.extraData[gameDataBundleKeys.game] = game
+        
+        // MARK: - Call Succeeded Closure
+        serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+          
+          guard let self = self else {
+            dispatchGroup.leave()
+            return
+          }
+          
+          guard let game = dataBundle.extraData[gameDataBundleKeys.game] as? Game else {
+            dispatchGroup.leave()
+            return
+          }
+          
+          self.decodeGameDetails(using: data, basedOn: game)
+          
+          dispatchGroup.leave()
+          
+        }
+        
+        // MARK: - ---------------------------
+
+        // MARK: - Call Failed Closure
+        serviceCaller.callFailed = { error in
+          dispatchGroup.leave()
+        }
+        
+        // MARK: - ---------------------------
+        
+        let urlString = "\(SteamURLComponents.specificGameURL)\(game.appID)"
+        let url: URL = URL(string: urlString)!
```

Comment:
Could we avoid force unwrapping here?

Reply:
That is debt that I missed, thanks for pointing that out